// !LANGUAGE: +MultiPlatformProjects
// MODULE: m1-common
// FILE: common.kt

<!DEPRECATED_MODIFIER_FOR_TARGET, JVM:DEPRECATED_MODIFIER_FOR_TARGET!>header<!> class My

<!DEPRECATED_MODIFIER_FOR_TARGET, JVM:DEPRECATED_MODIFIER_FOR_TARGET!>header<!> fun foo(): Int

<!DEPRECATED_MODIFIER_FOR_TARGET, JVM:DEPRECATED_MODIFIER_FOR_TARGET!>header<!> val x: String

<!DEPRECATED_MODIFIER_FOR_TARGET, JVM:DEPRECATED_MODIFIER_FOR_TARGET!>header<!> object O

<!DEPRECATED_MODIFIER_FOR_TARGET, JVM:DEPRECATED_MODIFIER_FOR_TARGET!>header<!> enum class E {
    FIRST
}

// MODULE: m2-jvm(m1-common)
// FILE: jvm.kt

<!DEPRECATED_MODIFIER_FOR_TARGET!>impl<!> class My

<!DEPRECATED_MODIFIER_FOR_TARGET!>impl<!> fun foo() = 42

<!DEPRECATED_MODIFIER_FOR_TARGET!>impl<!> val x get() = "Hello"

<!DEPRECATED_MODIFIER_FOR_TARGET!>impl<!> object O

<!DEPRECATED_MODIFIER_FOR_TARGET!>impl<!> enum class E {
    FIRST
}